org.eclipse.nebula.widgets.nattable.core/src/org/eclipse/nebula/widgets/nattable/ui/mode/ConfigurableModeEventHandler.java

```diff
@@ -59,13 +65,34 @@ public void mouseDown(MouseEvent event) {
         }
     }
 
+    private List<MouseMoveAction> currentActiveMoveActions = new ArrayList<>();
+
     @Override
     public synchronized void mouseMove(MouseEvent event) {
         if (event.x >= 0 && event.y >= 0) {
+            event.data = NatEventData.createInstanceFromEvent(event);
+
+            // check if current active move actions are still active
+            for (Iterator<MouseMoveAction> it = this.currentActiveMoveActions.iterator(); it.hasNext();) {
+                MouseMoveAction currentAction = it.next();
+                LabelStack regionLabels = this.natTable.getRegionLabelsByXY(event.x, event.y);
+                if (currentAction.mouseEventMatcher.matches(this.natTable, event, regionLabels) && currentAction.reexecuteEntryAction) {
+                    currentAction.run(this.natTable, event);
+                } else {
+                    currentAction.runExit(this.natTable, event);
+                    it.remove();
+                }
+            }
+
             IMouseAction mouseMoveAction = getUiBindingRegistry().getMouseMoveAction(event);
```

Comment:
Fetching only the "first" action from the registry is, of course, possible.  
By going that route, it still produces the issue that you have to define those events in a proper order in your code. If the "bigger matcher" (e.g. a cell) is first in the list, "the smaller matcher" (e.g. a painter in the cell) will never be reached, although the mouse cursor is on the painter. Sure, you can always add matchers in the required order, but for more complex setups this can be difficult. 
I'd go for querying all mouse move matchers which in effect also would eliminate the distinction between "register" and "registerFirst" in the UiBindingRegistry. What's the reason for you to not fetch all matching actions at once here? Hiding stuff? I'd go for deregistering in that case.

Reply:
The reason is that this is the design of ui bindings in NatTable. Why? I don't know, that decision was made before I joined the project. Probably because of performance reasons, and to provide the option to override the behavior without deregistering. I am not sure what would happen, if we simply change the long existing default behavior of mouse binding execution.

Reply:
I do agree. That's leave it like this (and it's a cool extension anyway)